packages/ui/src/ui/store/index.ts

```diff
@@ -32,3 +32,4 @@ export function createAppStore() {
 
 export type AppBrowserHistory = ReturnType<typeof createAppStore>['history'];
 export type StoreType = ReturnType<typeof createAppStore>['store'];
+export type AppThunkDispatch = ThunkDispatch<RootState, any, AnyAction>;
```

Comment:
Probably it is better to write separate function like `useAppDispatch`/`useThunkDispatch` and use it when we need to know type of returned value from an async action.